### PR TITLE
[MAINT] Format nilearn/glm: apply isort and fix flake8 errors

### PR DESCRIPTION
**Pull request by Remi-Gau**
Relates to #2528

Changes proposed in this pull request:

- [x] run isort
- [x] run pyupgrade
- [x] run flynt
- [x] fix remaining flake8 issues
- [x] update flake8 / isort CI

### CODE REVIEW

File: nilearn/glm/_base.py [hunk 1]
```diff
@@ -1,16 +1,16 @@
...
 from nibabel.onetime import auto_attr
+from nilearn._utils import CacheMixin
+from sklearn.base import BaseEstimator, TransformerMixin
 
 
 class BaseGLM(BaseEstimator, TransformerMixin, CacheMixin):
-    """Implementation of a base class for the
-    :term:`General Linear Model<GLM>`.
-    """
+    """Implement of a base class \
```
Comment (NicolasGensollen) [inline nilearn/glm/_base.py:7]:
```suggestion
    """Implement a base class \
```

**Comment by Remi-Gau** [comment]
@NicolasGensollen 
breaking the reformatting into 
- one PR for flake8 and isort 
- one PR for black
should make things a bit more digestable on the review side, no?

File: nilearn/glm/first_level/design_matrix.py [hunk 2]
```diff
@@ -166,18 +165,18 @@ def _make_drift(drift_model, frame_times, order, high_pass):
...
+        names.append(f'drift_{int(k)}')
     names.append('constant')
     return drift, names
 
 
 def _convolve_regressors(events, hrf_model, frame_times, fir_delays=[0],
                          min_onset=-24, oversampling=50):
-    """ Creation of  a matrix that comprises
-    the convolution of the conditions onset with a certain hrf model
+    """Creation of  a matrix that comprises \
```
Comment (NicolasGensollen) [inline nilearn/glm/first_level/design_matrix.py:178]:
```suggestion
    """Creation of a matrix that comprises \
```

File: nilearn/glm/first_level/hemodynamic_models.py [hunk 3]
```diff
@@ -85,7 +84,7 @@ def _gamma_difference_hrf(tr, oversampling=50, time_length=32., onset=0.,
 
 
 def spm_hrf(tr, oversampling=50, time_length=32., onset=0.):
-    """Implementation of the SPM hrf model
+    """Implement of the SPM hrf model.
```
Comment (NicolasGensollen) [inline nilearn/glm/first_level/hemodynamic_models.py:87]:
```suggestion
    """Implement the SPM hrf model.
```

File: nilearn/glm/first_level/hemodynamic_models.py [hunk 4]
```diff
@@ -111,7 +110,7 @@ def spm_hrf(tr, oversampling=50, time_length=32., onset=0.):
 
 
 def glover_hrf(tr, oversampling=50, time_length=32., onset=0.):
-    """Implementation of the Glover hrf model
+    """Implement of the Glover hrf model.
```
Comment (NicolasGensollen) [inline nilearn/glm/first_level/hemodynamic_models.py:113]:
```suggestion
    """Implement the Glover hrf model.
```

File: nilearn/glm/first_level/hemodynamic_models.py [hunk 5]
```diff
@@ -139,7 +138,7 @@ def glover_hrf(tr, oversampling=50, time_length=32., onset=0.):
 
 
 def spm_time_derivative(tr, oversampling=50, time_length=32., onset=0.):
-    """Implementation of the SPM time derivative hrf (dhrf) model
+    """Implement of the SPM time derivative hrf (dhrf) model.
```
Comment (NicolasGensollen) [inline nilearn/glm/first_level/hemodynamic_models.py:141]:
```suggestion
    """Implement the SPM time derivative hrf (dhrf) model.
```

File: nilearn/glm/first_level/hemodynamic_models.py [hunk 6]
```diff
@@ -170,7 +169,7 @@ def spm_time_derivative(tr, oversampling=50, time_length=32., onset=0.):
 
 
 def glover_time_derivative(tr, oversampling=50, time_length=32., onset=0.):
-    """Implementation of the Glover time derivative hrf (dhrf) model
+    """Implement of the Glover time derivative hrf (dhrf) model.
```
Comment (NicolasGensollen) [inline nilearn/glm/first_level/hemodynamic_models.py:172]:
```suggestion
    """Implement the Glover time derivative hrf (dhrf) model.
```

File: nilearn/glm/first_level/hemodynamic_models.py [hunk 7]
```diff
@@ -201,7 +200,7 @@ def glover_time_derivative(tr, oversampling=50, time_length=32., onset=0.):
 
 
 def spm_dispersion_derivative(tr, oversampling=50, time_length=32., onset=0.):
-    """Implementation of the SPM dispersion derivative hrf model
+    """Implement of the SPM dispersion derivative hrf model.
```
Comment (NicolasGensollen) [inline nilearn/glm/first_level/hemodynamic_models.py:203]:
```suggestion
    """Implement the SPM dispersion derivative hrf model.
```

File: nilearn/glm/first_level/hemodynamic_models.py [hunk 8]
```diff
@@ -233,7 +232,7 @@ def spm_dispersion_derivative(tr, oversampling=50, time_length=32., onset=0.):
 
 def glover_dispersion_derivative(tr, oversampling=50, time_length=32.,
                                  onset=0.):
-    """Implementation of the Glover dispersion derivative hrf model
+    """Implement of the Glover dispersion derivative hrf model.
```
Comment (NicolasGensollen) [inline nilearn/glm/first_level/hemodynamic_models.py:235]:
```suggestion
    """Implement the Glover dispersion derivative hrf model.
```

File: nilearn/glm/first_level/hemodynamic_models.py [hunk 9]
```diff
@@ -389,9 +388,12 @@ def _orthogonalize(X):
...
 def _regressor_names(con_name, hrf_model, fir_delays=None):
-    """ Returns a list of regressor names, computed from con-name and hrf type
-    when this information is explicitly given. If hrf_model is
-    a custom function or a list of custom functions, return their name.
+    """Return a list of regressor names, \
+    computed from con-name and hrf type \
+    when this information is explicitly given.
+
+    If hrf_model is a custom function or a list of custom functions,
+    return their name.
```
Comment (NicolasGensollen) [inline nilearn/glm/first_level/hemodynamic_models.py:396]:
```suggestion
    return their names.
```

File: nilearn/glm/second_level/second_level.py [hunk 10]
```diff
@@ -99,26 +100,26 @@ def _check_input_as_type(second_level_input,
...
 
 def _check_input_as_first_level_model(second_level_input,
                                       none_confounds):
     for model_idx, first_level in enumerate(second_level_input):
         if (first_level.labels_ is None or first_level.results_ is None):
-             raise ValueError(
-                 'Model %s at index %i has not been fit yet'
-                 '' % (first_level.subject_label, model_idx))
+            raise ValueError(
+                'Model %s at index %i has not been fit yet'
```
Comment (NicolasGensollen) [inline nilearn/glm/second_level/second_level.py:113]: why not an fstring here ?

**Review by NicolasGensollen — APPROVED** [review]
LGTM, thanks @Remi-Gau !
Just a few small suggestions.

File: nilearn/glm/first_level/first_level.py [hunk 11]
```diff
@@ -1,5 +1,4 @@
-"""
-This module contains the GLM and contrast classes that are meant to be the main
+"""Contain the GLM and contrast classes that are meant to be the main \
```
Comment (ymzayek) [inline nilearn/glm/first_level/first_level.py:1]:
```suggestion
"""Contains the GLM and contrast classes that are meant to be the main \
```

File: nilearn/glm/first_level/first_level.py [hunk 12]
```diff
@@ -74,7 +78,7 @@ def mean_scaling(Y, axis=0):
 
 
 def _ar_model_fit(X, val, Y):
-    """Wrapper for fit method of ARModel to allow joblib parallelization"""
+    """Wrap for fit method of ARModel to allow joblib parallelization."""
```
Comment (ymzayek) [inline nilearn/glm/first_level/first_level.py:81]:
```suggestion
    """Wrapper for fit method of ARModel to allow joblib parallelization."""
```

File: nilearn/glm/first_level/first_level.py [hunk 13]
```diff
@@ -158,9 +162,9 @@ def run_glm(Y, X, noise_model='ar1', bins=100,
     acceptable_noise_models = ['ols', 'arN']
     if ((noise_model[:2] != 'ar') and (noise_model != 'ols')):
         raise ValueError(
-            "Acceptable noise models are {0}. You provided "
-            "'noise_model={1}'".format(acceptable_noise_models,
-                                       noise_model)
+            "Acceptable noise models are {}. You provided "
+            "'noise_model={}'".format(acceptable_noise_models,
+                                      noise_model)
```
Comment (ymzayek) [inline nilearn/glm/first_level/first_level.py:167]: f-string

File: nilearn/glm/first_level/first_level.py [hunk 14]
```diff
@@ -176,7 +180,7 @@ def run_glm(Y, X, noise_model='ar1', bins=100,
 
         err_msg = ('AR order must be a positive integer specified as arN, '
                    'where N is an integer. E.g. ar3. '
-                   'You provided {0}.'.format(noise_model))
+                   'You provided {}.'.format(noise_model))
```
Comment (ymzayek) [inline nilearn/glm/first_level/first_level.py:183]: f-string

File: nilearn/glm/first_level/first_level.py [hunk 15]
```diff
@@ -231,8 +235,7 @@ def run_glm(Y, X, noise_model='ar1', bins=100,
 
 @fill_doc
 class FirstLevelModel(BaseGLM):
-    """ Implementation of the General Linear Model
-    for single session fMRI data.
+    """Implement of the General Linear Model for single session fMRI data.
```
Comment (ymzayek) [inline nilearn/glm/first_level/first_level.py:238]:
```suggestion
    """Implement the General Linear Model for single session fMRI data.
```

File: nilearn/glm/first_level/hemodynamic_models.py [hunk 16]
```diff
@@ -497,26 +499,26 @@ def _hrf_kernel(hrf_model, tr, oversampling=50, fir_delays=None):
...
     elif hrf_model is None:
         hkernel = [np.hstack((1, np.zeros(oversampling - 1)))]
     else:
-        raise ValueError('"{0}" is not a known hrf model. '
+        raise ValueError('"{}" is not a known hrf model. '
                          'Use either a custom model or '
-                         'one of {1}'.format(hrf_model,
-                                             acceptable_hrfs))
+                         'one of {}'.format(hrf_model,
+                                            acceptable_hrfs))
```
Comment (ymzayek) [inline nilearn/glm/first_level/hemodynamic_models.py:514]: f-string

File: nilearn/glm/first_level/design_matrix.py [hunk 17]
```diff
@@ -106,7 +105,7 @@ def _cosine_drift(high_pass, frame_times):
         warn('High-pass filter will span all accessible frequencies '
              'and saturate the design matrix. '
              'You may want to reduce the high_pass value.'
-             'The provided value is {0} Hz'.format(high_pass))
+             'The provided value is {} Hz'.format(high_pass))
```
[the remark this answers is not in the document]
Comment (ymzayek) [reply nilearn/glm/first_level/design_matrix.py:108]: Oh just saw your comment about f-strings. You can ignore these then!

File: nilearn/glm/first_level/first_level.py (hunk 12 above)
Comment (Remi-Gau) [reply nilearn/glm/first_level/first_level.py:81]: well flake8 will complain that this is not in the imperative mood, so maybe "create a wrapper" which is not ideal either.

suggestions are welcome

or we could silence that flake8 rule

Comment (ymzayek) [reply nilearn/glm/first_level/first_level.py:81]: Ok maybe just "Wrap fit method..."

File: nilearn/glm/first_level/first_level.py (hunk 11 above)
Comment (Remi-Gau) [reply nilearn/glm/first_level/first_level.py:1]: Will throw a flake8 error for not using the imperative mood.

Comment (Remi-Gau) [reply nilearn/glm/first_level/first_level.py:1]: Wait this is a module level docstring: this may actually work.

Comment (ymzayek) [reply nilearn/glm/first_level/first_level.py]: Ok, can also leave as is

Comment (Remi-Gau) [reply nilearn/glm/first_level/first_level.py]: It actually works: imperative mood is only for methods and functions doc string

**Review by bthirion — APPROVED** [review]
LGTM, thx.

**Comment by Remi-Gau** [comment]
Will merge this tomorrow unless I hear otherwise or get more change request.

**Comment by bthirion** [comment]
Sounds good. coverage is not great in this module, but this is stuff for another PR.

**Comment by Remi-Gau** [comment]
> Sounds good. coverage is not great in this module, but this is stuff for another PR.

totally agree !

my ideal plan for each "folder" is:

- make PEP8 compliant
- reformat with black
- refactor tests
- address most obvious blind post in coverage
- and ideally "stress-test" the test-suite through [mutation testing](https://en.wikipedia.org/wiki/Mutation_testing)

**Comment by Remi-Gau** [comment]
and this one goes in too
thanks @NicolasGensollen and @bthirion

**Pull request merged**